core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -47,19 +47,19 @@ object TransactionCoordinator {
       config.transactionTopicMinISR)
 
     val pidManager = new ProducerIdManager(config.brokerId, zkUtils)
-    val logManager = new TransactionStateManager(config.brokerId, zkUtils, scheduler, replicaManager, txnConfig, time)
+    val txnStateManager = new TransactionStateManager(config.brokerId, zkUtils, scheduler, replicaManager, txnConfig, time)
     val txnMarkerPurgatory = DelayedOperationPurgatory[DelayedTxnMarker]("txn-marker-purgatory", config.brokerId)
-    val transactionMarkerChannelManager = TransactionMarkerChannelManager(config, metrics, metadataCache, txnMarkerPurgatory, time)
+    val txnMarkerChannelManager = TransactionMarkerChannelManager(config, metrics, metadataCache, txnStateManager, txnMarkerPurgatory, time)
 
-    new TransactionCoordinator(config.brokerId, pidManager, logManager, transactionMarkerChannelManager, txnMarkerPurgatory, time)
+    new TransactionCoordinator(config.brokerId, pidManager, txnStateManager, txnMarkerChannelManager, txnMarkerPurgatory, time)
   }
 
   private def initTransactionError(error: Errors): InitPidResult = {
```

Comment:
I have also thought about that, e.g. having sth. like

```
case class InitPidResult(pid: Long = 
RecordBatch.NO_PRODUCER_ID, epoch: Short = RecordBatch.NO_PRODUCER_EPOCH, error: Errors)
```

and call it without overriding defaults upon checking failure cases, but I found it is less readable. If you feel strong about it I can change.